exchanges/bitstamp/bitstamp_test.go

```diff
@@ -181,198 +163,156 @@ func TestCalculateTradingFee(t *testing.T) {
-	}
-}
-
-func TestCancelAllExistingOrders(t *testing.T) {
-	t.Parallel()
-
-	_, err := b.CancelAllExistingOrders()
-	if err == nil {
-		t.Error("Test Failed - CancelAllExistingOrders() error", err)
-	}
-}
-
-func TestPlaceOrder(t *testing.T) {
-	t.Parallel()
-
-	if !areTestAPIKeysSet() {
-		t.Skip()
-	}
-
-	_, err := b.PlaceOrder("btcusd", 0.01, 1, true, true)
-	if err == nil {
-		t.Error("Test Failed - PlaceOrder() error")
+	switch {
+	case areTestAPIKeysSet() && err != nil && !mockTests:
+		t.Error("Test Failed - GetOrderStatus() error", err)
+	case !areTestAPIKeysSet() && err == nil && !mockTests:
+		t.Error("Expecting an error when no keys are set")
+	case mockTests && err == nil:
+		t.Error("Expecting an error until a QA pass can be completed")
 	}
 }
 
 func TestGetWithdrawalRequests(t *testing.T) {
-	t.Parallel()
+	TestSetup(t)
 
 	_, err := b.GetWithdrawalRequests(0)
-	if err == nil {
+	switch {
+	case areTestAPIKeysSet() && err != nil && !mockTests:
+		t.Error("Test Failed - GetWithdrawalRequests() error", err)
+	case !areTestAPIKeysSet() && err == nil && !mockTests:
+		t.Error("Expecting an error when no keys are set")
+	case mockTests && err != nil:
 		t.Error("Test Failed - GetWithdrawalRequests() error", err)
-	}
-	_, err = b.GetWithdrawalRequests(-1)
-	if err == nil {
-		t.Error("Test Failed - GetWithdrawalRequests() error")
-	}
-}
-
-func TestCryptoWithdrawal(t *testing.T) {
-	t.Parallel()
-
-	if !areTestAPIKeysSet() {
-		t.Skip()
-	}
-
-	_, err := b.CryptoWithdrawal(0, "bla", "btc", "", true)
-	if err == nil {
-		t.Error("Test Failed - CryptoWithdrawal() error", err)
-	}
-}
-
-func TestGetBitcoinDepositAddress(t *testing.T) {
-	t.Parallel()
-
-	_, err := b.GetCryptoDepositAddress(currency.BTC)
-	if err == nil {
-		t.Error("Test Failed - GetCryptoDepositAddress() error", err)
 	}
 }
 
 func TestGetUnconfirmedBitcoinDeposits(t *testing.T) {
-	t.Parallel()
+	TestSetup(t)
 
 	_, err := b.GetUnconfirmedBitcoinDeposits()
-	if err == nil {
+	switch {
+	case areTestAPIKeysSet() && err != nil && !mockTests:
+		t.Error("Test Failed - GetUnconfirmedBitcoinDeposits() error", err)
+	case !areTestAPIKeysSet() && err == nil && !mockTests:
+		t.Error("Expecting an error when no keys are set")
+	case mockTests && err != nil:
 		t.Error("Test Failed - GetUnconfirmedBitcoinDeposits() error", err)
 	}
 }
 
 func TestTransferAccountBalance(t *testing.T) {
-	t.Parallel()
+	TestSetup(t)
 
-	if !areTestAPIKeysSet() {
+	if !areTestAPIKeysSet() && !mockTests {
 		t.Skip()
 	}
 
-	_, err := b.TransferAccountBalance(1, "", "", true)
-	if err == nil {
+	err := b.TransferAccountBalance(0.01, "btc", "testAccount", true)
+	if !mockTests && err != nil {
 		t.Error("Test Failed - TransferAccountBalance() error", err)
 	}
-
-	_, err = b.TransferAccountBalance(1, "btc", "", false)
-	if err == nil {
-		t.Error("Test Failed - TransferAccountBalance() error", err)
+	if mockTests && err == nil {
+		t.Error("Expecting an error until a QA pass can be completed")
```

Comment:
The mock tests continue to run, I have recorded an error payload because I could not do an account balance transfer. Maybe I can run a t.Log just to be more verbose about the entire situation and remind that we need to record a correct response if we have a balance/correct api keys. Its also hard when they don't have up to date json responses in API documentation.

If there is no recorded response it will error but as I am doing every function one by one that will it will always be addressed.